source/common/tls/server_context_impl.cc

```diff
@@ -401,7 +401,9 @@ bool ServerContextImpl::isClientEcdsaCapable(const SSL_CLIENT_HELLO& ssl_client_
             CBS_len(&signature_algorithms_ext) != 0) {
           return false;
         }
-        if (cbsContainsU16(signature_algorithms, SSL_SIGN_ECDSA_SECP256R1_SHA256)) {
+        if (cbsContainsU16(signature_algorithms, SSL_SIGN_ECDSA_SECP256R1_SHA256) ||
```

Comment:
With support for multiple signature algorithms, using a bool for ecdsa-capable is insufficient. It would be more robust to match the signature algorithm of the cert to the supports algorithms of the client. 

For reference, I just looked at a random tls handshake I already had open in wireshark, and saw that the client supports p-256 and p-384, but does NOT support p-521. So with this code, if envoy were configured with a p-521 and and rsa cert, the handshake would fail when it should have succeeded.